plugins/plugin-chart-table/src/controlPanel.tsx

```diff
@@ -136,6 +140,23 @@ const config: ControlPanelConfig = {
             name: 'groupby',
             override: {
               visibility: isAggMode,
+              mapStateToProps: (state: ControlPanelState, controlState: ExtraControlProps) => {
+                const newState: ExtraControlProps = {};
+                if (state.datasource) {
+                  const options = state.datasource.columns.filter(c => c.groupby);
+                  if (controlState.includeTime) {
+                    options.unshift(TIME_COLUMN_OPTION);
+                  }
+                  newState.options = options;
+                }
+                newState.externalValidationErrors =
+                  ensureIsArray(state.controls.metrics?.value).length === 0 &&
+                  ensureIsArray(state.controls.percent_metrics?.value).length === 0 &&
+                  ensureIsArray(controlState.value).length === 0
+                    ? ['Group by, metrics or percentage metrics must have a value']
+                    : [];
+                return newState;
+              },
```

Comment:
I think it would be nice if we did the same check on `metrics` and `percent_metrics` to paint them red, too.

Reply:
Done. Check out the new video in PR description